osu.Game/Scoring/ScoreManager.cs

```diff
@@ -56,12 +56,44 @@ protected override ScoreInfo CreateModel(ArchiveReader archive)
             }
         }
 
+        public IEnumerable<ScoreInfo> GetScoresForBeatmapSet(int beatmapSetID, int rulesetID)
+        {
+            var scores = GetAllUsableScoresEnumerable()
+                .Where(s => !s.DeletePending && s.Beatmap.BeatmapSetInfoID == beatmapSetID && s.RulesetID == rulesetID);
+            scores = FilterByUser(scores);
+            return scores.ToList();
+        }
+
+        public IEnumerable<ScoreInfo> GetScoresForBeatmap(int beatmapID, int rulesetID)
+        {
+            var scores = GetAllUsableScoresEnumerable()
+                .Where(s => !s.DeletePending && s.BeatmapInfoID == beatmapID && s.RulesetID == rulesetID);
+            scores = FilterByUser(scores);
+            return scores.ToList();
+        }
+
+        public IQueryable<ScoreInfo> FilterByUser(IQueryable<ScoreInfo> scores)
```

Comment:
This is really confusing API design. A `ScoreManager` method, that *gets scores* from the outside, and uses the inner API provider (which up to now was used only to *download* scores), and then checks the *user*?

Should accept the *user* instead, not accept any scores, and *then* return a queryable.